prboom2/src/dsda/time.c

```diff
@@ -15,58 +15,37 @@
 //	DSDA Time
 //
 
+#ifdef _WIN32
+#include <windows.h>
+#else
 #include <time.h>
+#endif
 
 #include "i_system.h"
 
 #include "time.h"
 
-// [XA] MSVC hack: need to implement a handful of functions here.
-// this was yoinked from https://stackoverflow.com/a/31335254 so
-// no guarantees on if this stuff is accurate/kosher/whatever,
-// so maybe don't cut an official release with MSVC just yet.
-// it's enough to hack it into compiling for me, at least. ;)
-
-#ifdef _MSC_VER
-
-#include <windows.h>
+// lovey: Win32 interface
+#ifdef _WIN32
```

Comment:
Why `_WIN32` instead of `_MSC_VER`? In msys2 the unix code works and I'd like to keep it that way. What does _WIN32 evaluate to in that context?

Reply:
`_WIN32` because this is more efficient on windows, in both mingw and msvc.

If you look in the MinGW source ([https://sourceforge.net/projects/mingw-w64/files/mingw-w64/mingw-w64-release/mingw-w64-v10.0.0.zip/download](url), at mingw-w64-libraries/winpthreads/src/clock.c), `CLOCK_MONOTONIC` in `clock_gettime` is implemented with `QueryPerformanceFrequency` & `QueryPerformanceCounter`

Again, this was tested with MinGW x64, that was one of the first things I tested to make sure I didn't mess anything up.